aimmo-game/tests/test_simulation/test_turn_manager.py

```diff
@@ -182,5 +189,58 @@ def test_move_chain_fails_spiral(self):
         self.run_turn()
         [self.assert_at(avatars[i], locations[i]) for i in range(5)]
 
+    def build_test_by_constructor(self, constructor):
+        self.run_by_manager_turn(constructor)
+        self.run_by_manager_several_turns_and_avatars(constructor)
+        self.run_by_manager_several_turns(constructor)
+        self.run_by_manager_move_chain_fails_spiral(constructor)
+        self.run_by_manager_move_chain_fails_cycle(constructor)
+        self.run_by_manager_move_chain_fails_occupied(constructor)
+
+    def test_concurrent_turn_manager(self):
+        constructor = lambda x, y: self.construct_concurrent_turn_manager(x, y)
+        self.build_test_by_constructor(constructor)
+        self.run_by_manager_move_chain_fails_collision(constructor)
+        self.run_by_manager_move_chain_succeeds(constructor)
+
+    def sequential_move_chain_consecutive_avatars_fails(self):
+        '''
+        Given:  > > > > > _
+        Expect: _ o o o o o
+
+        This should fail for the sequential manager as the first avatar will bump into the second one
+        '''
+        self.construct_sequential_turn_manager([MoveEastDummy for _ in range(5)],
+                          [Location(x, 0) for x in range(5)])
+        avatars = [self.get_avatar(i) for i in range(5)]
+
+        [self.assert_at(avatars[x], Location(x, 0)) for x in range(5)]
+        self.run_turn()
+        [self.assert_at(avatars[x], Location(x, 0)) for x in range(4)]
+        self.assert_at(avatars[4], Location(5, 0))
+
+    def sequential_move_chain_fails_collission(self):
+        '''
+        Given:  > > > _ <
+        (1)
+        Expect: x x x _ x
+        '''
+        locations = [Location(0, 0), Location(1, 0), Location(2, 0), Location(4, 0)]
+        self.construct_sequential_turn_manager(
+            [MoveEastDummy, MoveEastDummy, MoveEastDummy, MoveWestDummy],
+            locations)
+        avatars = [self.get_avatar(i) for i in range(4)]
+
+        [self.assert_at(avatars[i], locations[i]) for i in range(4)]
+        self.run_turn()
+        [self.assert_at(avatars[i], locations[i]) for i in [0, 1, 3]]
+        self.assert_at(avatars[2], Location(3, 0))
+
+    def test_sequential_turn_manager(self):
+        constructor = lambda x, y: self.construct_sequential_turn_manager(x, y)
+        self.build_test_by_constructor(constructor)
+        self.sequential_move_chain_consecutive_avatars_fails()
+        self.sequential_move_chain_fails_collission()
```

Comment:
collision, not collission

Reply:
didn't it fail when run? ;)

Reply:
Doesn't fail because the method was called like that too. Whoever did it before me made the mistake twice. Fixed now!